hack/update-codegen-clients.sh

```diff
@@ -27,7 +27,42 @@ BOILERPLATE_HEADER="$( pwd )/hack/boilerplate/boilerplate.go.txt"
 popd
 CODEGEN_PKG=${CODEGEN_PKG:-$(cd "${SCRIPT_ROOT}"; go list -f '{{.Dir}}' -m k8s.io/code-generator)}
 
-bash "${CODEGEN_PKG}"/generate-groups.sh "deepcopy,client" \
+go install "${CODEGEN_PKG}"/cmd/applyconfiguration-gen
+
+"$GOPATH"/bin/applyconfiguration-gen \
+  --input-dirs github.com/kcp-dev/kcp/pkg/apis/core/v1alpha1 \
+  --input-dirs github.com/kcp-dev/kcp/pkg/apis/apiresource/v1alpha1 \
+  --input-dirs github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1 \
+  --input-dirs github.com/kcp-dev/kcp/pkg/apis/tenancy/v1beta1 \
+  --input-dirs github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1 \
+  --input-dirs github.com/kcp-dev/kcp/pkg/apis/scheduling/v1alpha1 \
+  --input-dirs github.com/kcp-dev/kcp/pkg/apis/topology/v1alpha1 \
+  --input-dirs github.com/kcp-dev/kcp/pkg/apis/workload/v1alpha1 \
+  --input-dirs github.com/kcp-dev/kcp/pkg/apis/third_party/conditions/apis/conditions/v1alpha1 \
+  --input-dirs k8s.io/apimachinery/pkg/apis/meta/v1,k8s.io/apimachinery/pkg/runtime,k8s.io/apimachinery/pkg/version,k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1 \
+  --output-package github.com/kcp-dev/kcp/pkg/client/applyconfiguration \
+  --go-header-file ./hack/../hack/boilerplate/boilerplate.generatego.txt \
+  --output-base "${SCRIPT_ROOT}" \
+  --trim-path-prefix github.com/kcp-dev/kcp
+
+"$GOPATH"/bin/client-gen \
```

Comment:
What does this call to `client-gen` do?

Reply:
The upstream `generate-groups.sh` script does not support passing the `apply-configuration-package` option to `client-gen`, so this call does what `generate-groups.sh "client"` would do, but with the extra `apply-configuration-package` passed to `client-gen`. Upstream `code-generator` scripts completely lack support for `applyconfiguration-gen`, so I went for the path of least resistance, and have `code-generator` improved asynchronously.

Reply:
Upstream PR: kubernetes/kubernetes#114987